Evaluate performance of covariates at predicting various mutations

dhimmel opened the pull request:
Creates an explore directory and README for this type of exploratory notebook.

See how well covariates (non-expression features) predict TP53 mutation.

Related to https://github.com/cognoma/machine-learning/issues/8: General mutation-load does provide some ability to predict mutation status of TP53.

Partially addresses https://github.com/cognoma/machine-learning/issues/21: Covariates are extracted from `samples.tsv`.

dhimmel committed:
Evaluate performance of covariates on TP53

Creates an explore directory and README for this type of exploratory notebook.

See how well covariates (non-expression features) predict TP53 mutation.

Related to https://github.com/cognoma/machine-learning/issues/8:
General mutation-load does provide some ability to predict mutation status of
TP53.

Partially addresses https://github.com/cognoma/machine-learning/issues/21:
Covariates are extracted from samples.tsv.

cgreene commented:
> Related to #8: General mutation-load does provide some ability to predict mutation status of TP53.

This isn't too surprising given TP53's role in controlling cell cycle checkpoints. Is this only true for genes in cell cycle checkpoint or DNA repair pathways, or is it also true for genes in proliferation pathways?

dhimmel committed:
Evaluate more covariate/mutation combinations

Evaluate covariate-only classifiers for the interesting mutations compiled in
https://github.com/cognoma/cancer-data/pull/22#issuecomment-245294155.

Switches to an expand grid system for evaluating all possible covariate
combinations.

Plot performance of all covariates on each mutation.

Switches to `covariates.tsv` created in
https://github.com/cognoma/cancer-data/pull/24 for encoded covariates.

dhimmel commented:
> Is this only true for genes in cell cycle checkpoint or DNA repair pathways, or is it also true for genes in proliferation pathways?

@cgreene, I updated the notebook to evaluate performance of covariate-only classifiers for the 8 interesting mutations we've [previously considered](https://github.com/cognoma/cancer-data/pull/22#issuecomment-245294155). Here is performance of the models with all covariates included:

![covariate-performance](https://cloud.githubusercontent.com/assets/1117703/18639994/1ebc17e8-7e64-11e6-8955-df34631c7266.png)

So actually, _TP53_ is among the hardest to predict using only covariates. _VHL_ which is highly disease-specific achieves a near-perfect AUROC. Therefore, without the disease or organ covariate, expression classifiers of _VHL_ are likely just classifying kidney clear cell carcinoma / kidney tissue.

See [this dataframe](https://github.com/dhimmel/machine-learning/blob/50aa17482dd1f9ebfae7d98fcf98a0209eb86487/explore/confounding/auroc.tsv) to get a general idea of covariate importance. Mutation load and disease type both seem important.

cgreene commented:
@dhimmel : Interesting! I can imagine that gene expression would clearly capture disease/organ, so if one usually had a mutation in a single gene that was relatively specific (e.g. VHL) I could imagine that creating a strong signal.

gwaybio commented on explore/confounding/confounding.py:
```diff
@@ -0,0 +1,280 @@
+
```
looks like you need to `nbconvert` this again

gwaybio replied:
I'll take a look again once it's easier to read

dhimmel replied:
Good catch. should be fixed now.

gwaybio reviewed (changes requested):
looks like nbconvert `confounding.ipynb` has to be rerun

dhimmel committed:
Export clean notebook to script

gwaybio commented on explore/README.md:
```diff
@@ -0,0 +1,9 @@
+# A directory for exploratory machine learning analyses
+
+This directory is home is exploratory analyses that help answer questions about how we should do machine learning. For algorithm implementations see the [`algorithms`](../algorithms) directory. For other types of analyses, place them here.
```
"This directory is home **to** exploratory analyses"?

gwaybio commented on explore/confounding/confounding.py:
```diff
@@ -0,0 +1,193 @@
+
+# coding: utf-8
+
+# # Create a logistic regression model to several mutations from covariates
```
"Create a logistic regression model to **predict** several mutations from covariates"?

gwaybio commented on explore/confounding/confounding.py:
```diff
@@ -0,0 +1,193 @@
+    return pd.DataFrame.from_records(rows, columns=data_dict.keys())
+
+mutations = {
+    '7157': 'TP53', # tumor protein p53
```
pep8 says inline comments should be separated by [two spaces](https://www.python.org/dev/peps/pep-0008/#inline-comments)

gwaybio commented on explore/confounding/confounding.py:
```diff
@@ -0,0 +1,193 @@
+# In[8]:
+
+def get_aurocs(X, y, series):
+    X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.1, random_state=0)
```
what does `series` look like?

gwaybio reviewed (changes requested):
Interesting and informative analysis. My comments are generally cosmetic, but it would be nice to get some additional documentation on some of the functions!

dhimmel committed:
Address review comments
